### PR TITLE
feat(auth-server): Convert verificationReminder templates to new stack

### PR DESCRIPTION
**Pull request by Yogita98**
## Because

- We have setup new templating stack in fxa auth server

## This pull request

- Converts verificationReminderFirst and verificationReminderSecond email templates to new stack

## Issue that this pull request solves

Closes: #9281 and #9285 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts:1]: The base template for both `verificationReminderFirst` and `verificationReminderSecond` emails remains the same so I created a separate partial `verificationReminder` for it and just passing down the templatevalues to it while rendering the emails

File: packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts:15]: Heh, I was going to suggest reducing the margin-bottom on this to better match the original template, and then I realized it's using an already established class. Yay for consistency!

I think we could perhaps reduce the padding on the first paragraph though. I think it makes sense to have a larger spacing between the header and text as you have it, as well as space between the last paragraph and button as you have it. 

The current/old template is showing `11px` margin-bottom here on the "confirmed it" paragraph:
image

Then in the new template, it's `20px`, coming from `primary-text`:
image

I think this should be `12px`. I don't think we should just slap `primary-text mb-3` on there though since it overrides the `mb` from `text-primary` and we'll probably need to use that class again.

 I'm kind of torn on what we should call it, maybe because I think `primary-text` isn't necessarily the best class name (I know we were just pulling from the old templates though). 

We _could_ create a class called `body-text` for now which can go on the first paragraph...

```
.body-text div {
  @extend .text-sm;
  @extend .mb-3;
  @extend .font-sans;
  text-align: center !important;
}
```
... though, then the class name "primary-text" is almost weirder. What makes it "primary"? Maybe later in a follow up we can rename these classes.

I'm OK with that for now, but there may be a better solution. I don't think we can do something like this:

```
.primary-text:last-of-type div {
  @extend .mb-5;
}

.primary-text div {
  @extend .text-sm;
  @extend .mb-3;
  @extend .font-sans;
  text-align: center !important;
}
```
...because of the weird DOM output MJML does. I originally thought we could just add `@extend .mt-5;` to the `primary-button` class and then set `text-primary` to `mb-3` and let collapsing margins work in our favor, but I don't think that works either because of the DOM output.

**Review by LZoog — COMMENTED** [review]
Looking good, all my comments are minor!

Is there any way we can also slap that `sans` class on here or something? Times New Roman hurts the eyes. 😆

image

File: packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts
Comment (Yogita98) [reply packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts:15]: Yeah I was actually working on refactoring the styles a bit in the `verify` template PR (still WIP) so I'll just move those changes in here.

File: packages/fxa-auth-server/lib/senders/emails/global.scss
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/global.scss:91]: We're actually not extending any of these margin classes anywhere except for `mb-5`. I think by keeping it as a separate class, we don't have to think about overriding styles anymore and can just add a separate class wherever we'd like

File: packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.ts:45]: Even I agree these classnames do look a little weird so it makes sense to assign them better names. Assuming there'd be more sections in the bodytext in future template conversions, I gave them names according to what underlying font size is being used and since it is used for `<mj-text>` elements, it made sense to start the classname with `text`

File: packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/partials/verificationReminder/index.ts:20]: This section now has the bottom margin of `12px` just like the old template

File: packages/fxa-auth-server/lib/senders/emails/global.scss
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/global.scss:70]: We could do this instead here:
```suggestion
    div, img {
      margin-bottom: $s-5 !important;
    }
```
... but I think we should just leave the extended `mb-5` on that sync logo instead. 🤔

File: packages/fxa-auth-server/lib/senders/emails/global.scss
Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/global.scss:91]: Hmm it might be a tad confusing to call this `text-xlarge` when we've got the `text-xl` class too.

I think we may want to avoid manually adding `mb` classes along with another class because it makes it easier to use the wrong spacing class which leads to inconsistency.  I don't mind a `text-header` class (that might be better than `header-text`) with your `.text-xlarge` styles + `mb-5` extended and then, use that class on the header at the top of every email, but I think that's basically what I recommended with the placeholder suggestion yesterday, heh.

BTW looks like this will happen anyway, but I'd say wait until your other PRs are merged so you can update these classes in those templates as well.

File: packages/fxa-auth-server/lib/senders/emails/global.scss
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/global.scss:103]: We can probably call this one `text-footer`. Since the color is grey, I don't think we can reuse this elsewhere.

File: packages/fxa-auth-server/lib/senders/emails/partials/cadReminder/index.ts
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/partials/cadReminder/index.ts:11]: Hmm yeah see my concern here is someone might just use `text-xlarge` and then like, add `mb-3` or `mb-4` here or on `sync-logo` or on anything else here and then lose that consistency we'd gain from it being in a single class.

File: packages/fxa-auth-server/lib/senders/emails/global.scss
Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/global.scss:91]: Maybe we can have the classes `text-header`, `text-body`, `text-body-mb-5`, and `text-footer`. I like all of these except the third one but am not sure what else to call it right now. 😆

Comment (Yogita98) [reply packages/fxa-auth-server/lib/senders/emails/global.scss:91]: Hmmm, right! But when I was working on converting `verify` email template, there was one more text section apart from header and body text and had styles essentially similar to the (new) `text-body` class but the font-color is darkish grey. See below

image

What do you think we should do in such situations? Not related to this PR ofc but we can potentially avoid refactoring the styles again in future PRs

Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/global.scss:91]: Huh, we could call it `text-body-grey` perhaps? Does it match the exact styles as `text-footer`?

Comment (Yogita98) [reply packages/fxa-auth-server/lib/senders/emails/global.scss:91]: I think the styles are similar to body text but yeah `text-body-grey` sounds good to me.

File: packages/fxa-auth-server/lib/senders/emails/partials/cadReminder/index.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/partials/cadReminder/index.ts:12]: Just realized this image is a little different than what we have in `fxa` layout so changed it's classname too

File: packages/fxa-auth-server/lib/senders/emails/templates/postVerify/index.ts
Comment (Yogita98) [inline packages/fxa-auth-server/lib/senders/emails/templates/postVerify/index.ts:11]: While refactoring the styles, I observed that the new template (left side) has a little more margin at the bottom after the first text section
image
We kinda missed it in the `postVerify` branch, but the new styles have taken care of that too

File: packages/fxa-auth-server/lib/senders/emails/global.scss
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/global.scss:93]: If this isn't a class we actually use, we can turn it into a placeholder (`%text-body-common`). 😁

File: packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/index.stories.ts
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/index.stories.ts:42]: Oh I'm sorry if I wasn't clear - I think we should use "default" if there's no other states, which is the case for both `verificationReminder` templates, but I don't think is the case for others. I think templates with two states, like this one being mobile and non-mobile, should have a description like how it was previously.

Comment (Yogita98) [reply packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/index.stories.ts:42]: I think we moved that mobile state from `cadReminder` template to `postVerify`. I had left a [comment](https://github.com/mozilla/fxa/pull/10120#discussion_r689717790) as well in the PR

Comment (Yogita98) [reply packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/index.stories.ts:42]: We actually removed the desktop state from `cadReminders` to `postVerify` template. I had also left a [comment](https://github.com/mozilla/fxa/pull/10120#discussion_r689717790) in that PR

Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/index.stories.ts:42]: Gotcha! I didn't realize we weren't sending `onDesktopOrTabletDevice` down on CAD so there's only one state.

**Review by LZoog — APPROVED** [review]
:shipit:

**Pull request merged**
